src/core/io.c

```diff
@@ -286,6 +316,9 @@ static Janet cfun_io_fclose(int32_t argc, Janet *argv) {
         janet_panic("file is closed");
     if (iof->flags & (JANET_FILE_NOT_CLOSEABLE))
         janet_panic("file not closable");
+    if (NULL != iof->buf) {
```

Comment:
The buffer should probably be freed _after_ the file is actually closed, since closing a file can cause it to flush, which references this buffer. As is, this causes UB.

Reply:
Yes, you're right. I missed that